File: lecture_6/assignements/sentimental-mario-more/mario.py
from cs50 import get_int


def main():
    height = get_valid_height()
    print_pyramid(height)


def get_valid_height():
    while True:
        height = get_int("Height: ")

        if 1 <= height <= 8:
            return height


def print_pyramid(n):
    for i in range(n):
        print(" " * (n - (i + 1)), end="")
        print("#" * (i + 1), end="")
        print("  ", end="")
        print("#" * (i + 1))


main()

File: lecture_0-2/assignements/mario-less/mario.c
#include <stdio.h>
#include <cs50.h>

void print_row(int spaces, int bricks);

int main(void)
{
    // Prompt the user for the pyramid's height
    int height;
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1);

    // Print a pyramid of that height
    for (int i = 0; i < height; i++)
    {
        // Print a pyramid row
        print_row(height - i - 1, i + 1);
    }
}

void print_row(int spaces, int bricks)
{
    // Print spaces (decreasing)
    for (int i = spaces; i > 0; i--)
    {
        printf(" ");
    }

    // Print blocks (increasing)
    for (int i = 0; i < bricks; i++)
    {
        printf("#");
    }
    printf("\n");
}
